### PR TITLE
Gci 708 get an order

### PR DESCRIPTION
**Pull request by nabeelamirch**
get an order

resolves: GCI-708

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/orders/api/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        this.orderService = orderService;
+    }
+
+    @GetMapping("${uk.gov.companieshouse.orders.api.orders}/{id}")
+    public ResponseEntity<OrderData> getOrder(final @PathVariable String id,
+                                              final @RequestHeader(REQUEST_ID_HEADER_NAME) String requestId) {
+        trace("ENTERING getOrder(" + id + ")", requestId);
+        final Order orderRetrieved = orderService.getOrder(id)
+                .orElseThrow(ResourceNotFoundException::new);
+        trace("EXITING getOrder(" + id + ") with" +orderRetrieved.getData(),  requestId);
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/orders/api/controller/OrderController.java]: Do we need a space after "with" here?

Comment (nabeelamirch) [reply src/main/java/uk/gov/companieshouse/orders/api/controller/OrderController.java]: yes we do, i'll add it in

File: src/test/java/uk/gov/companieshouse/orders/api/controller/OrderControllerIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    @Test
+    public void throwsNotFoundIfOrderDoesNotExist() throws Exception {
+        mockMvc.perform(get("/orders/"+ORDER_ID)
+                .header(REQUEST_ID_HEADER_NAME, TOKEN_REQUEST_ID_VALUE)
+                .header(ERIC_IDENTITY_HEADER_NAME, ERIC_IDENTITY_VALUE)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNotFound());
+    }
```
Comment (VaughanJackson123) [inline src/test/java/uk/gov/companieshouse/orders/api/controller/OrderControllerIntegrationTest.java]: I think the name `throwsNotFoundIfOrderDoesNotExist` is a little misleading. `respondsWith...` or something?

Comment (nabeelamirch) [reply src/test/java/uk/gov/companieshouse/orders/api/controller/OrderControllerIntegrationTest.java]: Yeh good point, "throws" implies that an exception is thrown

**Pull request merged**